clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -468,29 +490,30 @@ fn block_has_safety_comment(cx: &LateContext<'_>, span: Span) -> bool {
     // attributes and doc comments.
 
     matches!(
-        span_from_macro_expansion_has_safety_comment(cx, span),
-        HasSafetyComment::Yes(_)
-    ) || span_has_safety_comment(cx, span)
-}
-
-fn include_attrs_in_span(cx: &LateContext<'_>, hir_id: HirId, span: Span) -> Span {
-    span.to(cx.tcx.hir_attrs(hir_id).iter().fold(span, |acc, attr| {
-        if attr.is_doc_comment() {
-            return acc;
-        }
-        acc.to(attr.span())
-    }))
+        span_from_macro_expansion_has_safety_comment(cx, span, accept_comment_above_attributes),
+        HasSafetyComment::Yes(_, _)
+    ) || span_has_safety_comment(cx, span, accept_comment_above_attributes)
 }
 
+#[derive(Debug)]
 enum HasSafetyComment {
-    Yes(BytePos),
+    Yes(BytePos, bool),
     No,
     Maybe,
 }
 
+enum SafetyComment {
```

Comment:
Being that this enum is effectively equal to `HasSafetyComment`, is there a reason why they're separate?